src/detect.rs

```diff
@@ -18,3 +23,196 @@ pub enum DetectOutcome {
     Pass(BuildPlan),
     Fail,
 }
+
+enum BuildpackTomlArg<BM> {
+    Path(PathBuf),
+    Str(String),
+    BuildpackToml(BuildpackToml<BM>),
+}
+
+enum PlatformArg<P> {
+    Plat(P),
+    Dir(PathBuf),
+}
+
+pub struct DetectContextBuilder<P: Platform, BM: DeserializeOwned> {
+    app_dir: Option<PathBuf>,
+    buildpack_dir: Option<PathBuf>,
```

Comment:
We could model this without `Option` by creating an enum with two options: `Temporary` and `Static(PathBuf)`. In terms of code this is very similar to using an `Option`, but with additional semantics.

Reply:
Done. Again, great suggestion 🙏